src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,199 @@
+import DayPickerInput from 'react-day-picker/DayPickerInput';
+import { DateUtils, RangeModifier, DayPickerInputProps } from 'react-day-picker';
+import 'react-day-picker/lib/style.css';
+import dayjs from 'dayjs';
+
+import TextField from '../TextField';
+import Icon from '../Icon';
+import { wrapperStyle } from '../TextField/TextField.style';
+import { flex } from '../../theme/functions';
+import OverlayComponent from './OverlayComponent/OverlayComponent';
+
+export type Props = {
+  /** This boolean shows if the date picker will have the future dates available to select. Default: false */
+  disableFutureDates?: boolean;
+  /** This property is to define if this is a day picker or a day range picker */
+  isRangePicker?: boolean;
+  /** A callback to return user selection */
+  onChange?: (date: DateRange) => DateRange;
+};
+
+export type DateRange =
+  | RangeModifier
+  | {
+      from: undefined;
+      to: undefined;
+    };
+
+export type ExtraOption = { value: string; label: string; dates: Date | Date[] };
+
+const extraOptions: ExtraOption[] = [
+  {
+    value: 'last-7-days',
+    label: 'Last 7 days',
+    dates: [
+      dayjs()
+        .subtract(7, 'day')
+        .toDate(),
+      dayjs().toDate(),
+    ],
+  },
+  {
+    value: 'last-30-days',
+    label: 'Last 30 days',
+    dates: [
+      dayjs()
+        .subtract(30, 'day')
+        .toDate(),
+      dayjs().toDate(),
+    ],
+  },
+
+  {
+    value: 'custom',
+    label: 'Custom',
+    dates: dayjs().toDate(),
+  },
+];
+
+const DatePicker: React.FC<Props> = ({
+  disableFutureDates = false,
+  isRangePicker = false,
+  onChange,
+}) => {
+  const theme = useTheme();
+  const dayPickerRef = useRef<DayPickerInput>(null);
+  const daysInitialState = { from: undefined, to: undefined };
+  const [selectedOption, setSelectedOption] = useState();
+  const [selectedDay, setSelectedDay] = useState<DateRange>({ from: undefined, to: undefined });
+
+  useEffect(() => {
+    if (onChange) {
+      onChange(selectedDay);
+    }
+  }, [onChange, selectedDay]);
+
+  const handleDayRangeClick = useCallback(
+    day => {
+      const aboutToCompleteBothDates = selectedDay.from && !selectedDay.to;
+      const range = DateUtils.addDayToRange(
+        day,
+        // @TODO problem with the library as they define RangeModifiers as not null or undefined when they use it like this
+        // @ts-ignore
+        aboutToCompleteBothDates ? selectedDay : daysInitialState
+      );
+
+      if (aboutToCompleteBothDates) {
+        dayPickerRef.current?.hideDayPicker();
+      }
+
+      setSelectedDay(range);
+      setSelectedOption('custom');
+    },
+    [selectedDay, setSelectedDay, setSelectedOption, dayPickerRef.current]
+  );
+
+  const handleDayClick = useCallback(
+    day => {
+      setSelectedDay({ from: day, to: day });
+      setSelectedOption('custom');
+      dayPickerRef.current?.hideDayPicker();
+    },
+    [setSelectedDay, setSelectedOption, dayPickerRef.current]
```

Comment:
>ESLint: React Hook useCallback has an unnecessary dependency: 'dayPickerRef.current'. Either exclude it or remove the dependency array. Mutable values like 'dayPickerRef.current' aren't valid dependencies because mutating them doesn't re-render the component.(react-hooks/exhaustive-deps)

Reply:
I'd suggest you enable "Run `eslint --fix` on save", through Webstorm's settings.